runtime/resolvers/resolvers_test.go

```diff
@@ -0,0 +1,124 @@
+package resolvers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"maps"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/maxatome/go-testdeep/td"
+	"github.com/rilldata/rill/runtime"
+	"github.com/rilldata/rill/runtime/testruntime"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
+)
+
+type Project struct {
+	Sources    map[string]yaml.Node
+	Models     map[string]yaml.Node
+	Dashboards map[string]yaml.Node
+	APIs       map[string]yaml.Node
+}
```

Comment:
Instead of hard-coding the types, can we just use full paths as key types? We support quite a lot more resource types than just these four now. See this comment: 
https://github.com/rilldata/rill/issues/5218#issuecomment-2239087187

Reply:
I personally don't see much difference:
```
sources:
  1.yaml
  2.yaml
dashboards:
  1.yaml
  2.yaml
```

```
- sources/1.yaml:
- sources/2.yaml:
- dashboards/1.yaml:
- dashboards/2.yaml:
```

We can easily add a new field to Project struct if more resources is required. 
That could make sense if you'd like to process all files equally with a single for-loop and it can save a few lines of code but you'll have to repeat yourself if you need multiple similar-type entries in the future, ie:
```
- sources/1.yaml:
- sources/2.yaml:
- dashboards/1.yaml:
- dashboards/2.yaml:
```

Reply:
IMO reducing nesting one level for all the test files definitely improves readability here. The very slight repetition of the directory name does not hurt readabilty IMO, and anyway we don't require specific directory names anymore, so the files can just be created at the root level.

This also has the additional benefit of simplifying the test fixture parsing code.

Reply:
Done.